### PR TITLE
[DOC] Add DecoderMLP usage example

### PR DESCRIPTION
**Pull request by ZhangStudyLife**
#### Reference Issues/PRs

Partially addresses #2377.

This is intentionally limited to the v1 `DecoderMLP` model. #2378 covers DeepAR and NBeats.

#### What does this implement/fix?

This PR adds a focused, runnable usage example to the v1 `DecoderMLP` docstring. The example demonstrates:

- deterministic synthetic time-series data preparation;
- `TimeSeriesDataSet` construction;
- `DecoderMLP.from_dataset()`;
- one-batch CPU training with `fast_dev_run=True`;
- prediction with a stable output-shape check.

It also adds a minimal package-container linkage example, following the documentation structure used for the other v1 models.

The example does not download external data, require a GPU, or change model behavior.

#### What should a reviewer concentrate their feedback on?

- Whether the example is sufficiently concise while still showing the complete core workflow.
- Whether the package-container linkage example matches the intended v1 documentation pattern.

#### Did you add any tests for the change?

The examples are executable doctests. Local validation completed on Windows, Python 3.12, and CPU:

- direct doctests: 16 examples passed;
- `tests/test_models/test_mlp.py`: 7 passed;
- DecoderMLP estimator suite: 32 passed;
- applicable pre-commit hooks, Ruff, and Ruff formatting passed;
- full Sphinx HTML build passed with no DecoderMLP-specific warnings.

#### Any other comments?

The change is limited to two DecoderMLP documentation files. No model weights, datasets, generated files, or unrelated formatting changes are included.

Thank you for taking the time to review this contribution.

#### PR checklist

- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG].
- [x] Added/modified tests (the usage examples are executable doctests).
- [x] Used pre-commit hooks to ensure that the changed files comply with the configured hooks.

### CODE REVIEW

File: pytorch_forecasting/models/mlp/_decodermlp.py [hunk 1]
```diff
@@ -27,6 +27,53 @@ class DecoderMLP(BaseModelWithCovariates):
...
+    Examples
+    --------
+    Train on a small synthetic time series and make predictions:
+
+    >>> import pandas as pd
+    >>> import torch
+    >>> from lightning.pytorch import Trainer
+    >>> from pytorch_forecasting import DecoderMLP, TimeSeriesDataSet
+    >>> _ = torch.manual_seed(0)
+    >>> data = pd.DataFrame(
```
Comment (phoeenniixx) [inline pytorch_forecasting/models/mlp/_decodermlp.py:40]: you can also use [stallion](https://github.com/sktime/pytorch-forecasting/blob/7bcf66cbea5405a44a68a9265cad6bc164f70948/pytorch_forecasting/data/examples.py#L36) dataset here?

File: pytorch_forecasting/models/mlp/_decodermlp.py [hunk 2]
```diff
@@ -27,6 +27,53 @@ class DecoderMLP(BaseModelWithCovariates):
...
+
+    Examples
+    --------
+    Train on a small synthetic time series and make predictions:
+
+    >>> import pandas as pd
+    >>> import torch
+    >>> from lightning.pytorch import Trainer
+    >>> from pytorch_forecasting import DecoderMLP, TimeSeriesDataSet
+    >>> _ = torch.manual_seed(0)
```
Comment (phoeenniixx) [inline pytorch_forecasting/models/mlp/_decodermlp.py:39]: is there a specific reason for setting a seed here? I mean it is just an example, it doesnt need to be reproducible :)

**Review by phoeenniixx — CHANGES_REQUESTED** [review]
Thanks a lot and welcome to `pytorch-forecasting`! I think this is ready to merge, just a few doubts (see above)
